packages/symfony-toggle/test/DependencyInjection/FeatureRepositoryFactoryPassTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pheature\Test\Community\Symfony\DependencyInjection;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DriverManager;
+use Pheature\Community\Symfony\DependencyInjection\FeatureRepositoryFactoryPass;
+use Pheature\Core\Toggle\Write\FeatureRepository;
+use Pheature\Crud\Psr11\Toggle\ToggleConfig;
+use PHPUnit\Framework\TestCase;
+
+final class FeatureRepositoryFactoryPassTest extends TestCase
+{
+    public function testItShouldRegisterInMemoryFeatureRepositoryInContainer(): void
+    {
+        $compilerPass = new FeatureRepositoryFactoryPass();
+        $container = TestContainerFactory::create($compilerPass);
+        $container->register(ToggleConfig::class, ToggleConfig::class)->addArgument([
+            'driver' => 'inmemory',
+            'prefix' => '',
+        ]);
+
+        $featureRepositoryFactoryDefinition = $container->getDefinition(FeatureRepository::class);
+        self::assertFalse($featureRepositoryFactoryDefinition->isAutowired());
+        self::assertTrue($featureRepositoryFactoryDefinition->isLazy());
+
+        $featureRepositoryFactory = $container->get(FeatureRepository::class);
+        self::assertInstanceOf(FeatureRepository::class, $featureRepositoryFactory);
+    }
+
+    public function testItShouldRegisterDbalFeatureRepositoryInContainer(): void
+    {
+        touch('test.sqlite');
```

Comment:
If in the future can be a problem, we can use the [mocked filesystem](https://medium.com/weebly-engineering/phpunit-mocking-the-file-system-using-vfsstream-5d7d79b1eb2a) to virtualise the filesystem!

Reply:
Thanks!! I've created an issue to use the virtual filesystem ;-D #195